test/lib/mutation_source_test.cc

```diff
@@ -2558,12 +2558,10 @@ void for_each_schema_change(std::function<void(schema_ptr, const std::vector<mut
     }
 
     auto max_generator_count = std::max(
-        // boost::max_elements wants the iterators to be copy-assignable. The ones we get
-        // from boost::adaptors::transformed aren't.
-        boost::accumulate(static_columns | boost::adaptors::transformed([] (const column_description& c) {
+        std::ranges::fold_left(static_columns | std::views::transform([] (const column_description& c) {
             return c.data_generators.size();
         }), 0u, [] (size_t a, size_t b) { return std::max(a, b); }),
-        boost::accumulate(regular_columns | boost::adaptors::transformed([] (const column_description& c) {
+        std::ranges::fold_left(regular_columns | std::views::transform([] (const column_description& c) {
             return c.data_generators.size();
         }), 0u, [] (size_t a, size_t b) { return std::max(a, b); })
     );
```

Comment:
I would replace with:
```cpp
auto max_generator_count = std::ranges::max(
    std::ranges::max(
        static_columns | boost::adaptors::transformed([] (const column_description& c) {
            return c.data_generators.size();
        })
    ),
    std::ranges::max(
        regular_columns | boost::adaptors::transformed([] (const column_description& c) {
            return c.data_generators.size();
        })
    )
);
```

Reply:
what if `static_columns` is empty and / or `regular_columns` is empty? again, i'd prefer not take the risk of changing the existing behavior for no good reasons. this change is intended to be a refactor.

Reply:
1.
[If the range](https://en.cppreference.com/w/cpp/algorithm/ranges/max) is empty (as determined by [ranges::distance](http://en.cppreference.com/w/cpp/iterator/ranges/distance)(r)), the behavior is undefined.
Which means that we need to add a guard before calling `max`.

2.
I understand, and accept (to some extent) that this is a mechanical change and no other changes should be done.

3.
The way it is written now: fold-left by taking the maximum each time is so cumbersome that it might be good to change it at some point. It is much simpler to state what we do instead of doing it in an awkward way.

Reply:
well, this change is not a mechanical change. probably we can follow up with a change to switch some of the fold_left calls to `ranges::max()`. but for now, i prefer to keeping existing structure and focus on one thing and only one thing: migrate from boost::adaptors::transformed to std::views::transform.